### PR TITLE
Document Magento2 + Algolia setup

### PR DESCRIPTION
**Pull request by dpobel**
Related to https://gitlab.com/front-commerce/front-commerce/-/merge_requests/981

Main changes visible in:

* https://deploy-preview-303--heuristic-almeida-1a1f35.netlify.app/docs/magento2/search-engine.html#Algolia
* https://deploy-preview-303--heuristic-almeida-1a1f35.netlify.app/docs/appendices/migration-guides.html#2-12-0-gt-2-13-0

### CODE REVIEW

File: source/docs/magento2/search-engine.md [hunk 1]
```diff
@@ -70,3 +70,87 @@ query Search {
...
+
+```diff
+// .front-commerce.js
+-  modules: [],
++  modules: ["./node_modules/front-commerce/modules/datasource-algolia"],
+   serverModules: [
+     { name: "FrontCommerceCore", path: "server/modules/front-commerce-core" },
++    {
++      name: "Magento2Algolia",
++      path: "datasource-algolia/server/modules/magento1-algolia",
```
Comment (abailleulfc) [inline source/docs/magento2/search-engine.md:110]: There is a strange feeling here, we add "magento**1**-algolia" as the "Magento**2**Algolia" module

```suggestion
+      path: "datasource-algolia/server/modules/magento2-algolia",
```

File: source/docs/magento2/search-engine.md [hunk 2]
```diff
@@ -70,3 +70,87 @@ query Search {
...
+* the number of values per facet
+* the configured facet attributes
+
+On the configured facets, only the attribute name is taken into account (_Facet type_, _Label_, _Searchable_ and _Create Query rule_ are ignored for now).
+
+<blockquote class="warning">
+⚠️ For performance reason, the configuration retrieved from Magento is cached. As a result, after changing a parameter in the backoffice, the new parameter will be taken into account after at most one minute by Front-Commerce.
+</blockquote>
+
+After restarting Front-Commerce, you should be able run a GraphQL query to search for products, for instance:
```
Comment (abailleulfc) [inline source/docs/magento2/search-engine.md:134]:
```suggestion
After restarting Front-Commerce, you should be able to run a GraphQL query to search for products, for instance:
```

File: source/docs/magento2/search-engine.md [hunk 3]
```diff
@@ -70,3 +70,87 @@ query Search {
...
+      products {
+        sku
+        name
+      }
+    }
+  }
+}
+```
+
+If you are using the default theme or the theme Chocolatine, the search bar should now be visible.
```
Comment (abailleulfc) [inline source/docs/magento2/search-engine.md:151]:
```suggestion
If you are using the default theme or the Chocolatine theme, the search bar should now be visible.
```

**Review by abailleulfc — CHANGES_REQUESTED** [review]

File: source/docs/magento2/search-engine.md (hunk 1 above)
Comment (dpobel) [reply source/docs/magento2/search-engine.md]: fixed (integrated into 1f5a6d2)

File: source/docs/magento2/search-engine.md (hunk 2 above)
Comment (dpobel) [reply source/docs/magento2/search-engine.md]: fixed in 97d128d

File: source/docs/magento2/search-engine.md (hunk 3 above)
Comment (dpobel) [reply source/docs/magento2/search-engine.md]: fixed (integrated into 1f5a6d2)
also fixed in 97d128d for the M1 page

File: source/docs/appendices/migration-guides.md [hunk 4]
```diff
@@ -7,6 +7,23 @@ This area will contain the Migration steps to follow for upgrading your store to
 
 Our goal is to make migrations as smooth as possible. This is why we try to make many changes backward compatible by using deprecation warnings. The deprecation warnings are usually removed in the next breaking release.
 
+## `2.12.0` -> `2.13.0`
+
+### Upgrade the Magento2 module
+
+If you are using Magento2, version 2.6.0 of `front-commerce/magento2-module` is now the minimum required version. To update it to the last version, from Magento2 root, you can run:
```
Comment (real34) [inline source/docs/appendices/migration-guides.md:14]:
```suggestion
If you are using Magento2, version 2.6.0 of `front-commerce/magento2-module` is now the minimum required version. To update it to the latest version, from Magento2 root, you can run:
```

File: source/docs/magento2/search-engine.md [hunk 5]
```diff
@@ -70,3 +70,87 @@ query Search {
...
+1. under _Stores > Configuration > Algolia Search > Instant Search Result Page_, add `category_ids` in the _Facets_ parameter
+
+You can then run the indexer so that the products are indexed in Algolia's index.
+
+### Front-Commerce configuration
+
+First, you need to make sure the Algolia's search client is installed:
+
+```
+npm i algoliasearch
```
Comment (real34) [inline source/docs/magento2/search-engine.md:97]: **thought:** shouldn't we restrict it to a specific version here (so it matches the one [in optional dependencies](https://gitlab.com/front-commerce/front-commerce/-/blob/b3384a4e567080a0eae5d765f5f06c6be1a28213/package.json#L258))

```suggestion
npm i algoliasearch@4.8
```

**Review by real34 — APPROVED** [review]

**Comment by dpobel** [comment]
This MR was targetting main by mistake, the main commit has been reverted on
main and the branch manually merged into next

**Pull request merged**